operator/cmd/mesh/profile-common.go

```diff
@@ -50,13 +50,24 @@ func genIOPS(inFilename []string, profile, setOverlayYAML, ver string, force boo
 		return "", nil, fmt.Errorf("could not Unmarshal overlay Set%s: %s", setOverlayYAML, err)
 	}
 	if inFilename != nil {
-		b, err := ReadLayeredYAMLs(inFilename)
+		inputYaml, err := ReadLayeredYAMLs(inFilename)
 		if err != nil {
 			return "", nil, fmt.Errorf("could not read values from file %s: %s", inFilename, err)
 		}
-		overlayIOPS, overlayYAML, err = unmarshalAndValidateIOP(b, force)
+		overlayIOPS, overlayYAML, err = unmarshalAndValidateIOP(inputYaml, force)
 		if err != nil {
-			return "", nil, err
+			iopYAML, translateErr := translate.TranslateICPToIOPVer(inputYaml, binversion.OperatorBinaryVersion)
+			if translateErr != nil {
+				return "", nil, fmt.Errorf("could not unmarshal yaml or translate it to IOP: %s, %s\n\nOriginal YAML:\n%s",
+					err, translateErr, inputYaml)
+			}
+			l.logAndPrintf("%s\n\nIstio Operator CR has been upgraded. "+
+				"Your IstioControlPlane CR has been translated into IstioOperator CR above.\n"+
+				"Please keep the new IstioOperator CR for your future install or upgrade.", inputYaml)
```

Comment:
inputYaml should be iopYAML?

Reply:
This is intended to print the user's input when encountering an error, to help user identify the source of this error.

Reply:
sure, I will fix this.